### PR TITLE
[CI] Add Verify Linked Issue

### LINKED ISSUE #9397: Add CI for traceability
## Proposed refactoring or deprecation

The goal would be to start validating the PR description.

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

cc @borda @carmocca @akihironitta

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #9397

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananyahjha93 — APPROVED** [review]

**Comment by rohitgr7** [comment]
should we add a label check as well?
https://stackoverflow.com/a/62331521

**Review by justusschock — APPROVED** [review]

**Comment by tchaton** [comment]
> https://stackoverflow.com/a/62331521

We will improve this over time by filtering on some labels. But we are going through a label re-design. Worth to do after.

Best,
T.C

**Comment by Borda** [comment]
@tchaton is this ready? I do not see any checks in the description :rabbit:

File: .github/workflows/ci_verify_pr_contains_issue.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: VerifyIssue
+
+on:
+  pull_request:
+    types: [edited, synchronize, opened, reopened]
+  check_run:
```
Comment (Borda) [inline .github/workflows/ci_verify_pr_contains_issue.yaml:6]: why this one?
ca the PR change via any CI action? and if so pls write it in oomment :]

Comment (tchaton) [reply .github/workflows/ci_verify_pr_contains_issue.yaml:6]: I am following the template from this Github. I will work out other features from my fork this weekend

**Review by awaelchli — APPROVED** [review]
what happens if the PR does not link to an issue?

**Comment by tchaton** [comment]
> what happens if the PR does not link to an issue?

It will create a message like this one. I will play with my fork this weekend to customise this more. 

Screenshot 2021-09-15 at 08 01 25

**Review by kaushikb11 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
lets skip it for draft PRs

**Review by Borda — APPROVED** [review]
